### PR TITLE
feat(input): adds range to supported form inputs

### LINKED ISSUE #112: [feature request] Slider component (input type="range")
```
[ ] bug
[ x] feature request
[ ] enhancement
```

### 
input slider
I did try to search here on github and through the docs, but i could not find any. So i would like to voice my need for a slider component.

We have a design for a slider component here

image

### PR DESCRIPTION
**Pull request by hippee-lee**
- Adds tests for range controls in forms
- Adds docs for range input to website
- closes #112

This PR is and adoption and enhancement of the contribution from an external PR.
This also closes #3894.

Signed-off-by: Matt Hippely <mhippely@vmware.com>
Signed-off-by: Florin Borceanu <florin.borceanu@bytex.ro>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [x] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?
No built in support for range input controls in Clarity. 

Issue Number:  #112

## What is the new behavior?
Built in support for range input controls in Clarity forms
## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/clr-angular/forms/range/all.spec.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import RangeSpecs from './range.spec';
+import RangeContainerSpecs from './range-container.spec';
+
+fdescribe('Radio component', function() {
```
Comment (mathisscott) [inline src/clr-angular/forms/range/all.spec.ts]: Remove the `fdescribe`?

File: src/website/src/app/documentation/demos/range/range-basic.demo.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <clr-range-container [clrRangeHasProgress]="true">
+        <label>Value change example</label>
+        <input type="range" clrRange [(ngModel)]="input.two" name="three" (change)="rangeChange($event)"/>
+        <clr-control-helper>Helper text</clr-control-helper>
+    </clr-range-container>
+</form>
+
+<clr-tabs>
+    <clr-tab>
+        <button clrTabLink>Tab1</button>
```
Comment (mathisscott) [inline src/website/src/app/documentation/demos/range/range-basic.demo.html]: The title of both tabs is `Tab1`. Is this intentional? It looks weird in the docs.

File: src/clr-angular/forms/range/all.spec.ts (hunk 1 above)
Comment (hippee-lee) [reply src/clr-angular/forms/range/all.spec.ts]: Will remove.

File: src/website/src/app/documentation/demos/range/range-basic.demo.html (hunk 2 above)
Comment (hippee-lee) [reply src/website/src/app/documentation/demos/range/range-basic.demo.html]: Will update the tab text.

**Pull request merged**